src/openrc-run/openrc-run.c

```diff
@@ -448,9 +451,18 @@ svc_exec(const char *arg1, const char *arg2)
 				write_prefix(buffer, bytes, &prefixed);
 			}
 
-			/* Only SIGCHLD signals come down this pipe */
-			if (fd[0].revents & (POLLIN | POLLHUP))
+			/* Only SIGCHLD signals come down this pipe,
+			 * but with rc_parallel is might be another process */
+			if (fd[0].revents & (POLLIN | POLLHUP)) {
+				if (read(signal_pipe[0], &ret, sizeof(ret)) != sizeof(ret))
```

Comment:
Does it make sense to do this `read` in case of `POLLHUP` ?

Reply:
not really, but the code above with master_tty did the same (without even checking the return value...) so I just kept it simple because it also shouldn't hurt: read will just return 0 and not block in this case, so we'll print an error and bail -- I think that's as good as we can do.
I guess the error message won't make sense then, but it really shouldn't happen: HUP should only happen if `signal_pipe[1]` is closed and it's only closed by this thread after the loop ends... So I won't bother unless someone thinks it'll be useful.

Reply:
fixed comment and error message, should be possible to make the difference if we somehow get an impossible HUP without IN now (read 0 -> short read message)